gramps/gen/filters/rules/person/_hascommonancestorwith.py

```diff
@@ -96,17 +106,22 @@ def add_ancs(self, db, person):
     def reset(self):
         self.ancestor_cache = {}
 
-    def has_common_ancestor(self, other):
+    def has_common_ancestor(self, other: Person):
         for handle in self.with_people:
-            if (handle in self.ancestor_cache and self.ancestor_cache[handle]) & (
+            # Requires bitwise and for some reason:
+            left_and = (
+                handle in self.ancestor_cache and self.ancestor_cache[handle]
+            )  # type: ignore
```

Comment:
from reading the code, I would have expected this to be type `bool`
I'm assuming it's not since you've ignored the type?

Reply:
So, getting the type right is harder than you might think. To fix, one would need to get the types of the dependent variables correct, and would require a few changes. For example `set()` is not True, but the type is still `Set` (not `bool`), and `person` would need to change to `person is not None`, etc. If it is ok, I'll leave the type wrangling for a later PR. (The only reason the `ignore` is needed is because I added a `Person` type in the function signature, and I still want to pass the mypy checks).